### PR TITLE
Feat/sponsor 2

### PR DESCRIPTION
**Pull request by haydencleary**

### CODE REVIEW

File: app/p/[slug]/components/project-header/components/sponsor-button/sponsor-button.tsx [hunk 1]
```diff
@@ -13,28 +13,38 @@ import { TSponsorButton } from "./sponsor-button.types";
 
 export function SponsorButton({ project }: TSponsorButton.Props) {
   const { user } = useCurrentUser();
-  const [isOpen, setIsOpen] = useState(false);
+  const { sponsors } = user ?? {};
+  const [sponsor] = sponsors ?? [];
+  const { id: sponsorId } = sponsor ?? {};
+
+  const sponsorIdIsString = isString(sponsorId);
```
Comment (alexbeno) [inline app/p/[slug]/components/project-header/components/sponsor-button/sponsor-button.tsx:20]: typeof string ?

File: app/p/[slug]/components/project-header/components/sponsor-button/sponsor-button.tsx [hunk 2]
```diff
@@ -13,28 +13,38 @@ import { TSponsorButton } from "./sponsor-button.types";
...
+  const { data } = SponsorApi.queries.useGetSponsorById({
+    params: {
+      sponsorId: sponsorIdIsString ? sponsorId : "",
+    },
+    options: {
+      enabled: sponsorIdIsString,
+    },
+  });
+
+  const hasSponsorBudget = useMemo(() => (data?.availableBudgets.length ?? 0) > 0, [data]);
```
Comment (alexbeno) [inline app/p/[slug]/components/project-header/components/sponsor-button/sponsor-button.tsx:31]:
```suggestion
  const hasSponsorBudget = useMemo(() => !!data?.availableBudgets.length, [data]);
```

File: components/features/stacks/sponsor-project-stack/sponsor-project-stack.tsx [hunk 3]
```diff
@@ -1,37 +1,135 @@
...
+      await mutateAsync({
+        projectId: selectedProjectId,
+        amount: currencyAmountFloat,
+        currencyId: currencySelection?.id ?? "",
+      });
 
-    alert("Submit");
+      closeStack();
+    } catch (error) {
+      console.error(error);
```
Comment (alexbeno) [inline components/features/stacks/sponsor-project-stack/sponsor-project-stack.tsx:109]: should we keep the console ?

File: components/features/stacks/sponsor-project-stack/sponsor-project-stack.tsx [hunk 4]
```diff
@@ -46,73 +144,95 @@ export function SponsorProjectStack({ projectId }: TSponsorProjectStack.Props) {
...
+              {/*      color={balanceExceeded ? "orange" : currencyAmount ? "green" : undefined}*/}
+              {/*    />*/}
+              {/*    <Budget*/}
+              {/*      label={"v2.pages.stacks.sponsorProject.budget.budgetAfterAllocation"}*/}
+              {/*      // TODO @hayden selected project budget + currencyAmount*/}
+              {/*      amount={123}*/}
+              {/*      currency={currencySelection}*/}
+              {/*    />*/}
+              {/*  </ul>*/}
+              {/*</div>*/}
```
Comment (alexbeno) [inline components/features/stacks/sponsor-project-stack/sponsor-project-stack.tsx:219]: remove this ?

File: src/_pages/ProjectDetails/components/SponsorProjectButton.tsx [hunk 5]
```diff
@@ -15,35 +15,39 @@ interface Props {
...
+  const { data } = SponsorApi.queries.useGetSponsorById({
+    params: {
+      sponsorId: sponsorIdIsString ? sponsorId : "",
+    },
+    options: {
+      enabled: sponsorIdIsString,
+    },
+  });
+
+  const hasSponsorBudget = useMemo(() => (data?.availableBudgets.length ?? 0) > 0, [data]);
```
Comment (alexbeno) [inline src/_pages/ProjectDetails/components/SponsorProjectButton.tsx:34]:
```suggestion
  const hasSponsorBudget = useMemo(() => !!data?.availableBudgets.length, [data]);
```

File: components/features/stacks/sponsor-project-stack/sponsor-project-stack.tsx (hunk 4 above)
Comment (haydencleary) [reply components/features/stacks/sponsor-project-stack/sponsor-project-stack.tsx:219]: Need this for a coming feature

File: app/p/[slug]/components/project-header/components/sponsor-button/sponsor-button.tsx (hunk 2 above)
Comment (haydencleary) [reply app/p/[slug]/components/project-header/components/sponsor-button/sponsor-button.tsx]: 9d51dd3900c5fd60ac15d7daeb9fab9400a3bdba

File: src/_pages/ProjectDetails/components/SponsorProjectButton.tsx (hunk 5 above)
Comment (haydencleary) [reply src/_pages/ProjectDetails/components/SponsorProjectButton.tsx]: 9d51dd3900c5fd60ac15d7daeb9fab9400a3bdba

File: components/features/stacks/sponsor-project-stack/sponsor-project-stack.tsx (hunk 3 above)
Comment (haydencleary) [reply components/features/stacks/sponsor-project-stack/sponsor-project-stack.tsx]: b9f7ca5648108a9762847d19a68546f4d2328c06

**Review by alexbeno — APPROVED** [review]

**Pull request merged**